packages/plugins/typescript/resolvers/tests/mapping.spec.ts

```diff
@@ -332,23 +465,51 @@ describe('ResolversTypes', () => {
     )) as Types.ComplexPluginOutput;
 
     expect(result.prepend).toContain(`import { CustomPartial } from './my-wrapper';`);
+    // TODO: eddeee888 check if ResolversUnionTypes.MyUnion is supposed to be which?
+    //   - CustomPartial<Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> }>
+    //   OR
+    //   - CustomPartial<MyType>
+    expect(result.content).toBeSimilarStringTo(`
+      export type ResolversUnionTypes = {
+        ChildUnion: ( Omit<Child, 'parent'> & { parent?: Maybe<ResolversTypes['MyType']> } ) | ( MyOtherType );
+        MyUnion: ( CustomPartial<Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> }> ) | ( MyOtherType );
```

Comment:
This could be related to https://github.com/dotansimha/graphql-code-generator/issues/9043 as well.

Ideally, the `MyType` is should be treated the same way as `ResolversParentTypes['MyType']`. However:
- The `ResolversParentTypes['MyType']` is using the type from typescript plugin. 
- The mentioned issue seems to suggest that we should use the "Omit type" instead 

Currently, `ResolversUnionTypes` is using "Omit type" to replace the `{ T }` placeholder here but happy to change to match whichever direction `ResolversParentTypes['MyType']` uses.

cc @saihaj @gilgardosh

Reply:
FYI, according to #1631, the parent type should use the "Omit type", and #1677 implements it. Not sure when it broke between then and now, but I suspect it was unintentional.

Reply:
@eddeee888 awesome work!
I implemented a fix here #9110 , but it's conflicting. I'll adjust rebase it on top of yours after it's merged